### ISSUE TITLE
Run through Accessibility checklist

### ISSUE THREAD

**Issue by Jkrzy** [18F/samhsa-prototype#184]
## Let's run through the [18F accessibility checklist](https://accessibility.18f.gov/checklist/)

We'll make new issues for identified problems to track their remediation.

----
This checklist helps developers identify potential accessibility issues affecting their websites or applications. It's broken down into three sections of decreasing importance: A, B and C. Please check and address these issues in the order in which they appear.

For more detail on accessibility standards, please see [WCAG2.0 AA](https://www.w3.org/TR/WCAG20/) 

 * A - Critical issues that will cause serious problems and/or stop most users of assistive technology from using the site
 * B - Issues that may cause problems or increased frustration for certain users
 * C - Minor issues that will cause problems or frustration for a small number of users

It is important to note, while B and C are noted as less critical, they are still required to be truly 508 compliant. This checklist should be used as a reference for development and is not a substitute for compliance checks by a section 508 coordinator.

###  A - Critical
- [x] <a href='https://accessibility.18f.gov/keyboard/'>Site is keyboard accessible</a>
  * All interactions can be accessed with a keyboard
- [x] <a href='https://accessibility.18f.gov/keyboard/#keyboard-trap'>Site is free of keyboard traps</a>
  * The keyboard focus is never trapped in a loop
- [x] <a href='https://accessibility.18f.gov/forms/'>All `form` inputs have explicit labels</a>
- [x] <a href='https://accessibility.18f.gov/images/'>All relevant images use an `img` tag</a>
- [x] <a href='https://accessibility.18f.gov/images/'>All images have `alt` attributes</a>
- [x] <a href='https://accessibility.18f.gov/multimedia/'>Multimedia is tagged</a>
  * All multimedia has appropriate captioning and audio description
- [x] <a href='https://accessibility.18f.gov/color/'>Text has sufficient color contrast</a> DONE: #187 
  * All text has a contrast ratio of 4.5:1 with the background

### B - Less Critical

- [ ] <a href='https://accessibility.18f.gov/keyboard/'>Site never loses focus</a> TODO: #186 
  * Focus is always visible when moving through the page with the keyboard
- [x] <a href='https://accessibility.18f.gov/keyboard/'>Tab order is logical</a>
- [x] <a href='https://accessibility.18f.gov/forms/'>Form instructions are associated with inputs</a>
- [x] <a href='https://accessibility.18f.gov/timeouts/'>Site doesn't timeout unexpectedly</a>
  * Identify elements that may "timeout" and verify that the user can request more time
- [x] <a href='https://accessibility.18f.gov/tables/'>Tables are coded properly</a>
  * Tables have proper headers and column attributes
- [x] <a href='https://accessibility.18f.gov/headings/'>Headings are nested properly</a> DONE: #190 
  * Heading elements are nested in a logical way

### C - Minor
- [x] <a href='https://accessibility.18f.gov/iframes/'>Frames are named</a> DONE: #186 
  * All frames have a name element
- [x] <a href='https://accessibility.18f.gov/flashing/'>Flashing elements are compliant</a>
  * Elements that flash on screen do so at a rate of less than 3 Hz
- [x] <a href='https://accessibility.18f.gov/language/'>Language is set</a>
  * The language for the page is set
  * The language for sections on the page that differ from the site language are set
- [ ] <a href='https://accessibility.18f.gov/css/'>CSS is not required to use the page</a> TODO: #191 
  * The page makes sense with or without CSS
- [x] <a href='https://accessibility.18f.gov/links/'>Links are unique and contextual</a> DONE: #192 
  * All links can be understood taken alone, i.e 'Read more - about 508'
- [x] <a href='https://accessibility.18f.gov/page-titles/'>Page titles are descriptive</a> DONE: #185 
- [x] <a href='https://accessibility.18f.gov/plugins/'>Required plugins are linked on the page</a>

**Comment by Jkrzy** [comment]
Ran through https://accessibility.18f.gov/keyboard/ and was able to successfully navigate to all areas and functionality of the site.

**Comment by Jkrzy** [comment]
We have limited use of <img> tags on the site, all of which have associated `alt` attributes.
We have no multimedia.

The SAMHSA logo SVG does not have a `title` attribute, may need to add one.

**Comment by Jkrzy** [comment]
We have no tables, flashing elements, or timeouts.

**Comment by Jkrzy** [comment]
`lang` is set in `index.html`

**Comment by Jkrzy** [comment]
We're using a single page title across all views rendered by React. We should remediate and provide unique and descriptive titles for each page. https://accessibility.18f.gov/page-titles/

**Comment by Jkrzy** [comment]
The `iframe` in which google maps is rendered on the /results page does not have a title attribute. Can this be resolved on our end? https://accessibility.18f.gov/iframes/

**Comment by Jkrzy** [comment]
A few text color/contrast issues: https://accessibility.18f.gov/color/

Grey text on white background is 4:1 contrast ratio.
Results page information is 1.5:1
<img width="418" alt="Screen Shot 2019-08-05 at 4 45 32 PM" src="https://user-images.githubusercontent.com/3485564/62494094-aa38a480-b7a0-11e9-9c4a-7c40d179f172.png">
<img width="523" alt="Screen Shot 2019-08-05 at 4 45 36 PM" src="https://user-images.githubusercontent.com/3485564/62494127-be7ca180-b7a0-11e9-9746-d811a98c2fc8.png">
<img width="344" alt="Screen Shot 2019-08-05 at 4 48 36 PM" src="https://user-images.githubusercontent.com/3485564/62494209-e8ce5f00-b7a0-11e9-92bb-317017dbe546.png">

**Comment by Jkrzy** [comment]
Users should be able to tab once to select and move-past button links.

Most of the <button> elements on the site are wrapped by <Link> components. This renders as `<a><button></button></a>` which requires tabbing through both the anchor and the button element.

**Comment by Jkrzy** [comment]
Check and remediate headings for each page. 

The landing page has no <h1> and the footer has an out-of-order <h3> tag.

https://accessibility.18f.gov/headings/

**Comment by Jkrzy** [comment]
Improve site functionality when no CSS is present

When no CSS is present, keyboard navigation is broken. More than a few anchor tags are unreachable throughout the site and facility cards are unreachable on the results list page.

**Comment by Jkrzy** [comment]
For links which are not plainly and clearly described by their text, we should provide title or ARIA attributes.

The destination/functionality of links such as those on the site logo and content icons is unclear without the surrounding visual context.

https://accessibility.18f.gov/links/

**Comment by Jkrzy** [comment]
Superseded by new visual design work and accessibility assessments
